src/Uno.UI/UI/Xaml/Controls/Popup/Popup.iOS.cs

```diff
@@ -23,7 +24,9 @@ public UIView MainWindow
 			{
 				if (_mainWindow == null)
 				{
-					_mainWindow = UIApplication.SharedApplication.KeyWindow ?? UIApplication.SharedApplication.Windows[0];
+					_mainWindow = UIApplication.SharedApplication.KeyWindow
+							   ?? UIApplication.SharedApplication.Windows.FirstOrDefault()
+							   ?? throw new InvalidOperationException("Do not create popup until any app Window is created");
```

Comment:
It's generally better not to throw an exception from a getter (particularly this one that is for some reason public). Can you put the exception one step up in the call stack?

Reply:
Maybe you know a better way? Maybe we can subscribe to some event if UI has not loaded at the time of a call?

Reply:
Can you provide the stacktrace you're getting when this exception happens ?

Reply:
Create `ContentDialog` instance in App constructor. As I used `Host` I built DI in the constructor and put dialogs as dependencies. That caused them to be created before any window set.

Reply:
I would say this fix is fine, as it will provide a better exception for the user, but should not happen in most cases (when the API is used after initialization). I think we use a similar approach for APIs which need access to KeyWindow